subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/plugins/PrecompiledGroovyPluginsIntegrationTest.groovy

```diff
@@ -957,6 +958,55 @@ class PrecompiledGroovyPluginsIntegrationTest extends AbstractIntegrationSpec {
         succeeds(SAMPLE_TASK)
     }
 
+    def "should not allow precompiled plugin to conflict with core plugin"() {
+        given:
+        enablePrecompiledPluginsInBuildSrc()
+
+        file("buildSrc/src/main/groovy/plugins/java.gradle") << """
+            class TestTask extends DefaultTask {
+                @TaskAction
+                void run() {
+                    println 'from custom task'
+                }
+            }
+
+            task testTask(type: TestTask)
+        """
+
+        when:
+        def failure = fails "help"
+
+        then:
+        failure.assertHasCause("Precompiled plugin: 'java.gradle' conflicts with the core plugin: 'java'. " +
+            "See https://docs.gradle.org/" + GradleVersion.current().version + "/userguide/custom_plugins.html#sec:precompiled_plugins for more details.")
+    }
+
+    def "should not allow precompiled plugin to have org.gradle prefix"() {
+        given:
+        enablePrecompiledPluginsInBuildSrc()
+
+        file("buildSrc/src/main/groovy/plugins/${pluginName}.gradle") << """
+            class TestTask extends DefaultTask {
+                @TaskAction
+                void run() {
+                    println 'from custom task'
```

Comment:
nitpick: we don't verify the output; we can just remove the printline.

Reply:
Task is not even needed. Removed.